blogposts/announcing-sourcegraph-3.1.md

```diff
@@ -0,0 +1,100 @@
+
+Making configuration changes through the JSON editor is now easier too with easy to access documentation in the editor itself.
+
+<!-- TODO (ryan): GIF showing editor using the help tool tip with an item that requires a choice of values -->
+
+## Improved configuration management and documentation
+
+Sourcegraph 3.0 introduced the [management console](https://about.sourcegraph.com/blog/sourcegraph-3.0#management-console-and-site-config-improvements) for editing critical configuration to allow recovery from misconfiguration in the site admin.
+
+We've [updated our documentation](https://docs.sourcegraph.com/admin/config) to reflect this change, separating configuration into two sections:
+
+- [Critical configuration](https://docs.sourcegraph.com/admin/config/critical_config).
+- [Site configuration](https://docs.sourcegraph.com/admin/config/site_config).
+
+Reference documentation for [site configuration](https://docs.sourcegraph.com/admin/config/site_config#reference) and [critical configuration](https://docs.sourcegraph.com/admin/config/critical_config#reference) values are now rendered using the schema files themselves to ensure they'll never get out of date.
+
+## New Sourcegraph extension APIs
+
+The [Sourcegraph extension API](https://docs.sourcegraph.com/extensions/authoring) continues to evolve with two new APIs:
+
+- [Exposing text selection to extensions](https://github.com/sourcegraph/sourcegraph/issues/702).
+- [APIs for listening to active text editor changes and selections](https://github.com/sourcegraph/sourcegraph/issues/2026).
+
+This means the `[Git extras Sourcegraph extension](https://sourcegraph.com/extensions?query=git-extras) works better than ever. Want this functionality on GitHub too? Just install the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension).
+
+<!-- TODO (ryan): Gif of git-extras extension -->
+
+## Switching to scheduled releases on the 20th of each month
+
+We received feedback from many of you asking for a more regular and predictable release schedule and as a result, Sourcegraph will now be released on the 20th of each month 🎉.
+
+## Changelog
+
+See the [Sourcegraph changelog](https://github.com/sourcegraph/sourcegraph/blob/master/CHANGELOG.md#31) for a list of all changes in this release.
+
+## Thank you
+
+Thank you to the many people who contributed to make Sourcegraph better since our last release!
+
+<!-- TODO (ryan): Generate list -->
+
+<!--
+Use script at https://gist.github.com/ryan-blunden/c8898c989b48ba7a83e64ff9ae242cc2 to generate contributors for this release.
+
+If new Sourcegraphers have contributed, add their username to the exclude list.
+-->
+
+## Install or upgrade
+
+Ready to install? [Install Sourcegraph 3.1](https://docs.sourcegraph.com/)
+
+Upgrading from 2.x or 3.0.1? [See the migration guide](https://docs.sourcegraph.com/admin/migration/3_0)
```

Comment:
Hmm — I don't think 3.0.1 needs to follow this guide. Only 2.x and 3.0.0, right @nicksnyder?

Reply:
Correct